classes/UpgradeSelfCheck.php

```diff
@@ -141,15 +135,70 @@ class UpgradeSelfCheck
      */
     private $prestashopConfiguration;
 
+    /**
+     * @var DistributionApiService
+     */
+    private $distributionApiService;
+
+    const PRESTASHOP_17_PHP_REQUIREMENTS = [
+        '1.7.0' => [
+            'php_min_version' => '5.4',
+            'php_max_version' => '7.1',
+        ],
+        '1.7.1' => [
+            'php_min_version' => '5.4',
+            'php_max_version' => '7.1',
+        ],
+        '1.7.2' => [
+            'php_min_version' => '5.4',
+            'php_max_version' => '7.1',
+        ],
+        '1.7.3' => [
+            'php_min_version' => '5.4',
+            'php_max_version' => '7.1',
+        ],
+        '1.7.4' => [
+            'php_min_version' => '5.6',
+            'php_max_version' => '7.1',
+        ],
+        '1.7.5' => [
+            'php_min_version' => '5.6',
+            'php_max_version' => '7.2',
+        ],
+        '1.7.6' => [
+            'php_min_version' => '5.6',
+            'php_max_version' => '7.2',
+        ],
+        '1.7.7' => [
+            'php_min_version' => '5.6',
+            'php_max_version' => '7.3',
+        ],
+        '1.7.8' => [
+            'php_min_version' => '5.6',
+            'php_max_version' => '7.4',
+        ],
+    ];
+
+    const PHP_REQUIREMENTS_INVALID = 0;
+    const PHP_REQUIREMENTS_VALID = 1;
+    const UNKNOWN_PHP_REQUIREMENTS = 2;
```

Comment:
```suggestion
    const PHP_REQUIREMENTS_UNKNOWN = 2;
```

This would allow PHPStan to better check the output of a method of this class.